undertone-htb.js

```diff
@@ -306,84 +310,116 @@ function UndertoneHtb(configs) {
          *
          */
 
-        /* ---------- Process adResponse and extract the bids into the bids array ------------*/
-
         var adResponseDic = {};
-        if (adResponse != null) {
-            for (var i=0; i<adResponse.length; i++) {
-                adResponseDic[adResponse[i].bidRequestId] = adResponse[i];
+        if (adResponse !== null) {
+            for (var idx = 0; idx < adResponse.length; idx++) {
+                adResponseDic[adResponse[idx].bidRequestId] = adResponse[idx];
             }
         }
 
         /* MRA partners receive only one parcel in the array. */
-        for (var i = 0; i < returnParcels.length; i++) {
-            var returnParcel = returnParcels[i];
+        for (var parcelIdx = 0; parcelIdx < returnParcels.length; parcelIdx++) {
+            var returnParcel = returnParcels[parcelIdx];
 
-            /* header stats information */
-            var headerStatsInfo = {
-                sessionId: sessionId,
-                statsId: __profile.statsId,
-                htSlotId: returnParcel.htSlot.getId(),
-                xSlotNames: [returnParcel.xSlotName]
-            };
+            /* Header stats information */
+            var headerStatsInfo = {};
+            var htSlotId = returnParcel.htSlot.getId();
+            headerStatsInfo[htSlotId] = {};
+            headerStatsInfo[htSlotId][returnParcel.requestId] = [returnParcel.xSlotName];
 
             returnParcel.targetingType = 'slot';
             returnParcel.targeting = {};
-            returnParcel.targeting[__baseClass._configs.targetingKeys.id] = [returnParcel.requestId];
             returnParcel.pass = false;
 
-            var currAdResponse = adResponseDic[returnParcel.requestId];
-            if (currAdResponse == null) {
+            var currAdResponse = adResponseDic[returnParcel.xSlotRef.bidId];
+            if (currAdResponse === null) {
                 currAdResponse = {};
             }
 
-            if (currAdResponse.ad == null || currAdResponse.cpm <= 0) {
+            if (currAdResponse.ad === null || currAdResponse.cpm <= 0) {
                 if (__profile.enabledAnalytics.requestTime) {
                     __baseClass._emitStatsEvent(sessionId, 'hs_slot_pass', headerStatsInfo);
                 }
                 returnParcel.pass = true;
+
+                continue;
             }
 
-            returnParcel.adm = currAdResponse.ad || "";
-            returnParcel.price = currAdResponse.cpm || 0;
+            var bidCreative = currAdResponse.ad || '';
+            var bidPrice = currAdResponse.cpm || 0;
+            var bidDealId = '';
+            var pixelUrl = '';
 
-            if (currAdResponse.width != null) {
-                returnParcel.size = [currAdResponse.width, currAdResponse.height];
+            if (currAdResponse.width !== null) {
```

Comment:
The `null` here needs to be updated to `undefined` to ensure we are comparing the type and value appropriately.

Reply:
I can't do that because when there's your lint check does not allow to write undefined

Reply:
Please use typeof `currAdResponse.width === 'undefined'` instead then.

Reply:
when I do that it makes system tests fail for some reason, also fails when I check both null and undefined

Reply:
We were able to reproduce and identify why this was failing. Please update the ad property in  getValidResponse from `ad: creative` to `ad: creative || '<div id="1"></div>'`

Reply:
ok it solved one of two system tests failures.
Still fails this test:
System Tests: Partner Open Market Bids should log headerstats events correctly
Expected 1 to be 2.

Reply:
Did we update both instances of `ad: creative` ?

Reply:
You're right, fixed them both and now it works, changes pushed